### PR TITLE
Dont pass no_gpg_check in zypper_ar on SLE11

### PR DESCRIPTION
**Pull request by mimi1vx**
SLE11 zypper have limited funcionality :(

### CODE REVIEW

**Comment by pevik** [comment]
Generally LGTM. Suggestions (it's up to you whether to consider them):

* I'd split these into 2 as they're not related
* "SLE11 zypper have limited funcionality :(": is it QA repos related? (I suppose so, otherwise you'd put it into `zypper_ar`). Could you be a bit specific.
* comments you give to PR should be in commit message (nit, but why to expect people to look into PR instead of git commit?)

File: lib/repo_tools.pm [hunk 1]
```diff
@@ -56,8 +57,9 @@ Helper to add QA web repository repository.
 This repository is *not* mandatory.
 =cut
 sub add_qa_web_repo {
-    my $repo = get_var('QA_WEB_REPO');
-    zypper_ar($repo, name => 'qa-web', no_gpg_check => 1) if ($repo);
+    my $repo   = get_var('QA_WEB_REPO');
+    my $usable = is_sle("<12") ? 0 : 1;
+    zypper_ar($repo, name => 'qa-web', no_gpg_check => $usable) if ($repo);
```
Comment (pevik) [inline lib/repo_tools.pm]: Use `is_sle("<12")` directly (without `$usable` might be good enough for single use, but it's up to you (as `$usable` does not describe it much anyway).

Comment (baierjan) [reply lib/repo_tools.pm]: He could actually use `zypper_ar($repo, name => 'qa-web', no_gpg_check => is_sle('<12') ? undef : 1)` to overcome the issue. Still, the fix for the `zypper_ar` is nice to have :smile:

File: lib/utils.pm [hunk 2]
```diff
@@ -476,17 +476,17 @@ C<$params> other ar subcommand parameters, optional
...
     my $name         = $args{name}         // '';
-    my $priority     = $args{priority}     // '';
+    my $priority     = $args{priority}     // undef;
     my $params       = $args{params}       // '';
     my $no_gpg_check = $args{no_gpg_check} // '';
 
-    $priority     = "-p $priority"  if ($priority);
-    $no_gpg_check = "--no-gpgcheck" if ($no_gpg_check);
+    $priority     = defined($priority) ? "-p $priority"  : "";
+    $no_gpg_check = $no_gpg_check      ? "--no-gpgcheck" : "";
```
Comment (pevik) [inline lib/utils.pm]: I'm blind, don't see what is wrong on previous version. Some warning I overlooked?

Comment (mimi1vx) [reply lib/utils.pm]: try use `zypper_ar(... , no_gpg_check => 0)` :D

Comment (baierjan) [reply lib/utils.pm]: if `$priority == 0`: there will be no `-p 0` as the condition will be false
if `$no_gpg_check == 0` (false): the value (in this case `0`) will be added to the zypper string

Comment (pevik) [reply lib/utils.pm]: OK, didn't know that `0` is valid value as priority (I'll read man better next time). Thx for fix.

Comment (baierjan) [reply lib/utils.pm]: Even in the opposite case, using `priority => 0` will end up as `"--gpg-auto-import-keys ar -f 0 $no_gpg_check $params $url"` (same case as the gpg_check).

Comment (mimi1vx) [reply lib/utils.pm]: man zypper: `-p 0 will set the priority back to the default (99)`

**Comment by pevik** [comment]
Definitely not an expert on zypper. Looking into git, `--no-gpgcheck` option was added in https://github.com/openSUSE/zypper/commit/6609ee95569a94ef584f951f2f8131cfe1b6de4c (zypper 1.6.2) => SuSE-SLE-11-SP3-Branch branch (+ according to bnc#524787 then fixed in zypper-1.6.15). If we still test older SLES versions IMHO this should go to `zypper_ar` itself (+ some warning "ignored option due old SLES version).

**Comment by mimi1vx** [comment]
@pevik yes, unfortunately, oldest supported product is SLE11SP1TD which receives new kernels

**Pull request merged**